src/BloomExe/Book/XMatterHelper.cs

```diff
@@ -204,22 +204,33 @@ private void CleanupBrandingImages(XmlElement newPageDiv, string branding, strin
 			if (BookStorage.IsStaticContent(bookFolderPath))
 				return;
 			var prefix = BrandingApi.kApiBrandingImage + "?id=";
-			foreach (XmlElement imageElt in newPageDiv.SafeSelectNodes("//img"))
+			foreach (XmlElement imageElt in newPageDiv.SafeSelectNodes("//img").Cast<XmlElement>().ToArray())
 			{
 				var src = imageElt.Attributes["src"]?.Value;
 				if (src == null || !src.StartsWith(prefix))
 					continue;
 				var fileName = src.Substring(prefix.Length);
 				var pathToRealImage = BrandingApi.FindBrandingImageFileIfPossible(branding, fileName);
-				if (!string.IsNullOrEmpty(pathToRealImage))
+				if (string.IsNullOrEmpty(pathToRealImage))
+				{
+					// If the book folder contains this file already, it's obsolete, from some previous branding choice.
+					// Get rid of it to save space. We might also have an obsolete file with a png extension; get rid of
+					// that too.
+					var destFileName = Path.Combine(bookFolderPath, fileName);
+					RobustFile.Delete(destFileName);
+					RobustFile.Delete(Path.ChangeExtension(destFileName, ".png"));
+					// And we may as well get rid of the img element itself
```

Comment:
Can you say more than "might as well" here? It's a rare thing in code that there isn't some cost, especially when fixing something that isn't broken. Perhaps this even adds to complexity, since we rebuild these, so it's... it's not going to stay gone. But what if we later allow some branding, say, in a footer of content pages. Well now, this deletion will actually have a permanent, unexpected consequence.